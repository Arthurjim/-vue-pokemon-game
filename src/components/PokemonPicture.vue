<template>
    <div class="pokemon-container">

        <img
          v-bind:src="imgSrc"
          alt="pokemon"
          class="hidden-pokemon" />
        <img
            v-bind:src="imgSrc"
            v-if="showPokemon"
            alt="pokemon"
            class="fade-in"
        />
    </div>
</template>

<script>
export default {
    name:'PokemonPicture',
    props: {
        pokemonId: {
            type: Number,
            required:true,
        },
        showPokemon:{
            type:Boolean,
            required:true,
            default:false
        }
    },
   
    computed: {
        imgSrc() {
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
        },
    },
};
</script>

<style scoped>
.pokemon-container {
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
}
img {
    position: absolute;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>
