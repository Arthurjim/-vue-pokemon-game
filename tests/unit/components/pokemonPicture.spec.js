import PokemonPicture from "@/components/PokemonPicture.vue";
import { shallowMount } from "@vue/test-utils";
describe("Pruebas en el componente pokemonPicture", () => {
    test('Debe de hacer match con el snaptchot', () => { 
        const wrapper = shallowMount(PokemonPicture, {
            propsData: {
                pokemonId:1,
                showPokemon:true
                }
            })
        expect(wrapper.html()).toMatchSnapshot()
     })
     test('Debe de mostrar la imagen oculta y el pokemon 100', () => { 

        const wrapper = shallowMount(PokemonPicture, {
            propsData: {
                pokemonId:100,
                showPokemon:false
                }
            })
            console.log(wrapper.html())
            const [img1,img2]=wrapper.findAll('img')
            expect(img1.exists()).toBe(true)
            expect(img2).toBe(undefined)

            expect(img1.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
            // expect(img1.attributes.src).toBe('https://pokeres.bastionbot.org/images/pokemon/100.png')
            // expect(wrapper.find('.hidden-pokemon').exists()).toBe(false)

            // expect(wrapper.find('.fade-in').exists()).toBe(true)
      })

     test('debe de mostrar el pokemons si showPokemon:true', () => { 

        const wrapper = shallowMount(PokemonPicture, {
            propsData: {
                pokemonId:100,
                showPokemon:true
                }
            })
            const img1=wrapper.find('img')
            expect(img1.exists()).toBe(true)
            expect(img1.classes('fade-in')).toBe(true)
      })
});
