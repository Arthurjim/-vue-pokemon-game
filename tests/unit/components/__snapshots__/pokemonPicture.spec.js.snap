// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente pokemonPicture Debe de hacer match con el snaptchot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="fade-in" alt="pokemon"></div>`;
